flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -1971,7 +1967,7 @@ private <OUT, IN1, IN2> TypeInformation<OUT> privateGetForClass(
         }
 
         // special case for POJOs generated by Avro.
-        if (hasSuperclass(clazz, AVRO_SPECIFIC_RECORD_BASE_CLASS)) {
+        if (AvroUtils.isAvroSpecificRecord(clazz)) {
```

Comment:
I am wondering does does the avro-maven-plugin ever generate / support a POJO that inherits from a POJO class that implements  AVRO_SPECIFIC_RECORD_BASE_CLASS. If so should be walk the inheritance tree to find AVRO_SPECIFIC_RECORD_BASE_CLASS?

Reply:
Maybe you're right. I've noticed that the generated User class is not marked as final. One case I can think of is creating a non-generated class B that directly extends the generated class A. However, this is not within the scope of the changes in this PR. If you agree, I can create a new Jira ticket to track it.

Reply:
@xuyangzhong raising a new Jira to work this separately makes sense to me.

Reply:
Trace it at https://issues.apache.org/jira/browse/FLINK-36337